docs/ARCHITECTURE.md

```diff
@@ -24,79 +24,388 @@ Before delving into the details, let's understand a few key concepts that will b
+
+A summarized list of risks:
+
+#### Funds theft
+
+Funds can be permanently stolen if:
+
+- Our contracts receive a malicious or defective upgrade
+- Malicious L2 activity is undetected for more than 1 week
+
+#### Funds loss
+
+Funds can be permantely lost if:
+
+- Our contracts receive a defective upgrade
+
+#### Chain halt
+
+The chain might be halted as preemptive measure against the risks above. The chain will be halted if
+
+- Malicious activity is detected
+- Unintended behaviour of the L2 is detected
+
+This might temporarily lock user funds and cause inconvenience.
+
+### SPoFs
 
-Easing the operation and participation of light clients in the network without compromising on security is a key principle in Fuel. In that sense, Fuel strives to minimize the size of the information that is exchanged between peers through the protocol. One of the design decisions furthering that goal is the use of 64 bits (`u64`) to code the balances that are bridged from L1 to L2. The most popular token standard, `ERC20`, codes the amounts with `256 bits`. If the amount transferred from L1 to L2 surpasses the capacity of Fuel to mint the L2 counterpart, a `refund` message will be generated, that will allow the user to retrieve the originally depositted amount in L1.
+As it can be derived from the diagrams above, there are entities performing mission critical tasks:
 
-Additionally, it is a de-facto practice in the space to use 18 decimals to represent the token, whereas in Fuel, 9 decimals are used to represent amounts under the unit. This means that there is a potential loss of precision when it comes to translating L1 amounts to L2 amounts, as the user could lose some dust amounts between bridging operations. Ideally, this limitation could be worked around as long as the deposits initiated in L1 do not specify amounts greater than 9 decimals (i.e. 1.0000000010 does not lose amounts, 1.0000000011 would lose 0.0000000001). DApps that enable bridge operations should observe this limitation and truncate the amounts to avoid loss of precision.
+- Security council and smart contract ownership: the smart contracts will be managed by a security council (by means of a multisig) that enables key administration functions, such as upgrades, granting of roles and permissions for privileged smart contract functions, pausing of the system, etc. It is of the essence that the security council operates correctly, honestly and timely in the management of the smart contracts.
+- Fuel Blockchain Sequencer / Validator: The Fuel blockchain 's operation is currently a Proof of Authority scheme under which a single private key is able to build new blocks. A compromise on this key could mean the generation of rogue blocks. While it is needed to compromise more than just the private key of the PoA node in order to cause permanent damage, it would halt the network for an unknown amount of time.
```

Comment:
```suggestion
- Fuel Blockchain Sequencer / Validator: The Fuel blockchain's operation is currently a Proof of Authority scheme under which a single private key is able to build new blocks. A compromise on this key could mean the generation of rogue blocks. While it is needed to compromise more than just the private key of the PoA node in order to cause permanent damage, it would halt the network for an unknown amount of time.
```